tests/test_utils.py

```diff
@@ -179,39 +179,26 @@ def test_get_repository_config_missing(tmpdir):
     assert utils.get_repository_from_config(pypirc, "pypi") == exp
 
 
-def test_get_repository_config_invalid_scheme(tmpdir):
-    """Raise an exception for a URL with an invalid scheme."""
-    pypirc = os.path.join(str(tmpdir), ".pypirc")
-
-    with pytest.raises(
-        exceptions.UnreachableRepositoryURLDetected,
-        match=r"Invalid repository URL: "
-        r"scheme was required to be one of \['http', 'https'\] but was 'ftp'.",
-    ):
-        utils.get_repository_from_config(pypirc, "foo.bar", "ftp://test.pypi.org")
-
-
-def test_get_repository_config_missing_components(tmpdir):
-    """Raise an exception for a URL with a missing host and/or scheme."""
+@pytest.mark.parametrize(
+    "repo_url, message",
+    [
+        (
+            "ftp://test.pypi.org",
+            r"scheme was required to be one of \['http', 'https'\]",
+        ),
+        ("https:/", "host was required but missing."),
+        ("//test.pypi.org", "scheme was required but missing."),
+        ("foo.bar", "host, scheme were required but missing."),
+    ],
+)
+def test_get_repository_config_missing_components(tmpdir, repo_url, message):
```

Comment:
I think we can just parametrize this test which the parameters except the first one, and not remove `test_get_repository_config_invalid_scheme`, since that test looks for an `invalid` scheme, while this one looks for a `missing` scheme, which IMO are two different behaviours which should warrant separate tests.

Reply:
If you feel strongly about that, I'm game to make that change, but I'd rather leave it as-is, because it's the same path through the `rfc3986` code, raising the same exception with a different message:

https://github.com/pypa/twine/blob/194eccdde19a2064c3e4a22e7d9364bf9fcf7992/twine/utils.py#L103-L117

However, I did change the test name to be more general.

Reply:
I don't feel strongly about it. Your current name change is perfectly fine.